source/read/specify-a-query.txt

```diff
@@ -0,0 +1,274 @@
+
+.. tip::
+
+    To learn more about array operators, see the :manual:`Array Query Operators
+    </reference/operator/query-array/>` guide in the {+mdb-server+} manual.
+
+The following example specifies an array operator in a query filter as a
+parameter to the ``find`` method. The code returns all documents in which the
+``type`` array field contains ``2`` elements:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/read/specify-queries.rb
+      :start-after: start-find-array
+      :end-before: end-find-array
+      :language: ruby
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      {"_id"=>1, "name"=>"apples", "qty"=>5, "rating"=>3, "color"=>"red", "type"=>["fuji", "honeycrisp"]}
+      {"_id"=>3, "name"=>"oranges", "qty"=>6, "rating"=>2, "type"=>["naval", "mandarin"]}
+
+Element Operators
+-----------------
+
+Element operators query data based on the presence or type of a field. The following
+list describes each element operator: 
+
+- ``$exists``: Returns documents that contain the specified field
+- ``$type``: Returns documents that contain a field of the specified type
+
+.. tip::
+
+    To learn more about element operators, see the :manual:`Element Query Operators
+    </reference/operator/query-element/>` guide in the {+mdb-server+} manual.
+
+The following example specifies an element operator in a query filter as a
+parameter to the ``find`` method. The code returns all documents that have a
+``color`` field:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/read/specify-queries.rb
+      :start-after: start-find-element
+      :end-before: end-find-element
+      :language: ruby
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      {"_id"=>1, "name"=>"apples", "qty"=>5, "rating"=>3, "color"=>"red", "type"=>["fuji", "honeycrisp"]}
+      {"_id"=>2, "name"=>"bananas", "qty"=>7, "rating"=>4, "color"=>"yellow", "type"=>["cavendish"]}
+      {"_id"=>4, "name"=>"pineapples", "qty"=>3, "rating"=>5, "color"=>"yellow"}
+
+Evaluation Operators
+--------------------
+
+Evaluation operators return data based on evaluations of either individual
+fields or the entire collection's documents. The following list describes
+common element operators: 
+
+- ``$text``: Performs a text search on the documents
+- ``$regex``: Returns documents that match a specified regular expression
+- ``$mod``: Performs a modulo operation on the value of a field and
+  returns documents where the remainder is a specified value
+
+.. tip::
+
+    To view a full list of evaluation operators, see the :manual:`Evaluation Query Operators
+    </reference/operator/query-evaluation/>` guide in the {+mdb-server+} manual.
+
+The following example specifies an evaluation operator in a query filter as a
+parameter to the ``find`` method. The code uses a regular expression to return
+all documents in which the ``name`` field value has at least two consecutive
+``'p'`` characters:
```

Comment:
It's a bit odd to me that this section is about evaluation operators, and yet the example does not appear to use any evaluation operators (because it is depending on an implicit transformation). Is it worth mentioning that the Ruby driver will automatically convert `foo: /regex/` to `foo: { '$regex' => /regex/ }`?

Reply:
Yeah, definitely think that's good to mention. I added a note below the code example